ott/core/segment.py

```diff
@@ -76,15 +77,28 @@ def segment_point_cloud(
 
   segmented_a = []
   segmented_x = []
-  x = jnp.concatenate((x, jnp.zeros((1, dim))))
+
+  if custom_padder is None:
+    x = jnp.concatenate((x, jnp.zeros((1, dim))))
```

Comment:
This line adds the "zero" vector to the original matrix of points. This is done so because when idx below == num+1 (line 88) this gets the "zero" from x. I will add a comment to clarify this. 

Therefore, I think it will be sufficient to add that zero only once, in this location, and not recompute it in the loop several times as below. As a result, I do not think custom_padder should to be a callable. I think it can be called custom_padding and be a jnp.ndarray, and we see if the user passes it. When set to None, it can take the dimension of x and be a vector of zeros by default, as done in the original file in l.79.

Reply:
just to clarify: in this case, you would envision the user passing the padding explicitly? aka that he/she would have to know how to appropriately pad according to the cost?